components/sharing-dialog/src/constants.js

```diff
@@ -19,7 +19,254 @@ export const SHARE_TARGET_GROUP = 'SHARE_TARGET_GROUP'
  * Sharing dialog types
  */
 
-export const VISUALIZATION = 'visualization'
+export const ACCESS = 'access'
```

Comment:
Some of these are not `shareable`, is this just all schemas?

When I do `schemas.filter(s => s.shareable).map(s => s.name).sort()`, I get the following list:

```
['aggregateDataExchange', 'apiToken', 'attribute', 'category', 'categoryCombo', 'categoryOption', 'categoryOptionGroup', 'categoryOptionGroupSet', 'constant', 'dashboard', 'dataApprovalLevel', 'dataApprovalWorkflow', 'dataElement', 'dataElementGroup', 'dataElementGroupSet', 'dataSet', 'document', 'eventChart', 'eventFilter', 'eventReport', 'eventVisualization', 'expressionDimensionItem', 'externalMapLayer', 'identifiableObject', 'identifiableObject', 'identifiableObject', 'indicator', 'indicatorGroup', 'indicatorGroupSet', 'interpretation', 'legendSet', 'map', 'optionGroup', 'optionGroupSet', 'optionSet', 'organisationUnitGroup', 'organisationUnitGroupSet', 'predictorGroup', 'program', 'programIndicator', 'programIndicatorGroup', 'programStage', 'programStageWorkingList', 'relationshipType', 'report', 'sqlView', 'trackedEntityAttribute', 'trackedEntityFilter', 'trackedEntityType', 'userGroup', 'userRole', 'validationRule', 'validationRuleGroup', 'visualization']
```

Reply:
Would it make sense to group these in an object, then do ```Object.values(DIALOG_TYPES) in propTypes?```. I realize this changes the "structure" of the constants, so would need to check if this breaks something.
I also realized eg. `tabbed-content.js` may need an update to refer to the new array instead of each constant separately. 

Here's my generated object. (I copy pasted the schemas object from the request from https://debug.dhis2.org/dev/dhis-web-maintenance/index.html#/list/all , then put than in console and filtered with code above and transformed to an object).

```
const DIALOG_TYPES = {
    "AGGREGATE_DATA_EXCHANGE": "aggregateDataExchange",
    "API_TOKEN": "apiToken",
    "ATTRIBUTE": "attribute",
    "CATEGORY": "category",
    "CATEGORY_COMBO": "categoryCombo",
    "CATEGORY_OPTION": "categoryOption",
    "CATEGORY_OPTION_GROUP": "categoryOptionGroup",
    "CATEGORY_OPTION_GROUP_SET": "categoryOptionGroupSet",
    "CONSTANT": "constant",
    "DASHBOARD": "dashboard",
    "DATA_APPROVAL_LEVEL": "dataApprovalLevel",
    "DATA_APPROVAL_WORKFLOW": "dataApprovalWorkflow",
    "DATA_ELEMENT": "dataElement",
    "DATA_ELEMENT_GROUP": "dataElementGroup",
    "DATA_ELEMENT_GROUP_SET": "dataElementGroupSet",
    "DATA_SET": "dataSet",
    "DOCUMENT": "document",
    "EVENT_CHART": "eventChart",
    "EVENT_FILTER": "eventFilter",
    "EVENT_REPORT": "eventReport",
    "EVENT_VISUALIZATION": "eventVisualization",
    "EXPRESSION_DIMENSION_ITEM": "expressionDimensionItem",
    "EXTERNAL_MAP_LAYER": "externalMapLayer",
    "IDENTIFIABLE_OBJECT": "identifiableObject",
    "INDICATOR": "indicator",
    "INDICATOR_GROUP": "indicatorGroup",
    "INDICATOR_GROUP_SET": "indicatorGroupSet",
    "INTERPRETATION": "interpretation",
    "LEGEND_SET": "legendSet",
    "MAP": "map",
    "OPTION_GROUP": "optionGroup",
    "OPTION_GROUP_SET": "optionGroupSet",
    "OPTION_SET": "optionSet",
    "ORGANISATION_UNIT_GROUP": "organisationUnitGroup",
    "ORGANISATION_UNIT_GROUP_SET": "organisationUnitGroupSet",
    "PREDICTOR_GROUP": "predictorGroup",
    "PROGRAM": "program",
    "PROGRAM_INDICATOR": "programIndicator",
    "PROGRAM_INDICATOR_GROUP": "programIndicatorGroup",
    "PROGRAM_STAGE": "programStage",
    "PROGRAM_STAGE_WORKING_LIST": "programStageWorkingList",
    "RELATIONSHIP_TYPE": "relationshipType",
    "REPORT": "report",
    "SQL_VIEW": "sqlView",
    "TRACKED_ENTITY_ATTRIBUTE": "trackedEntityAttribute",
    "TRACKED_ENTITY_FILTER": "trackedEntityFilter",
    "TRACKED_ENTITY_TYPE": "trackedEntityType",
    "USER_GROUP": "userGroup",
    "USER_ROLE": "userRole",
    "VALIDATION_RULE": "validationRule",
    "VALIDATION_RULE_GROUP": "validationRuleGroup",
    "VISUALIZATION": "visualization"
}
// could also export in constants
const DIALOG_TYPES_LIST = Object.values(DIALOG_TYPES)

```

Reply:
Ah thanks @Birkbjo , definately just printed all the available schemas in the backend :) 
Re the object vs list of constants, yes I think that works as well, so happy to copy and paste from what you have in here :)